apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/java/selections/CompiledSelectionsBuilder.kt

```diff
@@ -68,7 +68,8 @@ class CompiledSelectionsBuilder(
   }
 
   private fun List<GQLSelection>.walk(name: String, private: Boolean, parentType: String): List<FieldSpec> {
-    val propertyName = resolveNameClashes(usedNames, context.layout.propertyName(name))
+    val modelName = if (name == root) name else context.layout.compiledSelectionsName(name)
```

Comment:
Nit: you could rename `private` to `isRoot` and use that instead of doing a string comparison here